### PR TITLE
Add option to MatrixWorkspaceDisplay to overplot bins

### LINKED ISSUE #30371: Workbench: Cannot overplot bins easily
As a user I have a plot across spectra along a single bin. I would like to overplot further bins on the existing plot that has been created. This can currently be done by scripting but the option is useful at the GUI level and was present in MantidPlot.

This task is to investigate how to best incorporate the bin overplotting without too much confusion in the plotting menu and implement the solution.

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR adds the option to overplot bin data in the matrix workspace display.

**To test:**

1. Load some data
2. Right click and show data
3. Select one of the columns and right click -> plot bin 
4. If you now select another column and right click you should be able to overplot this new bin.
5. Test this functionality and make sure it doesn't produce any unexpected behaviour.

Fixes #30371 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/widgets/workspacedisplay/matrix/presenter.py [hunk 1]
```diff
@@ -189,20 +189,34 @@ def _do_action_plot(self, table, axis, get_index, plot_errors=False):
...
+                  overplot=overplot, plot_kwargs=plot_kwargs)
 
     def action_plot_spectrum(self, table):
         self._do_action_plot(table, MantidAxType.SPECTRUM, lambda index: index.row())
 
     def action_plot_spectrum_with_errors(self, table):
         self._do_action_plot(table, MantidAxType.SPECTRUM, lambda index: index.row(), plot_errors=True)
 
+    def action_overplot_spectrum(self, table):
+        self._do_overaction_plot(table, MantidAxType.SPECTRUM, lambda index: index.row(),
```
Comment (Matt-Cumber) [inline qt/python/mantidqt/widgets/workspacedisplay/matrix/presenter.py:201]: I'm a bit confused with this line as _do_overaction_plot doesn't exist yet from what I've tested it works fine. Does this mean this is not called at all?

**Review by Matt-Cumber — CHANGES_REQUESTED** [review]
Code changes look good and work well. Just one comment

File: qt/python/mantidqt/widgets/workspacedisplay/matrix/presenter.py (hunk 1 above)
Comment (StephenSmith25) [reply qt/python/mantidqt/widgets/workspacedisplay/matrix/presenter.py:201]: Yeah I'm also confused. This a typo 🤦‍♂️ . But also wondering why its working.....I'll have a look in the afternoon

**Review by Matt-Cumber — APPROVED** [review]
Looks good to me, nice job 👍

**Pull request merged**